cpp/src/arrow/compute/exec.cc

```diff
@@ -325,35 +325,164 @@ bool ExecBatchIterator::Next(ExecBatch* batch) {
   return true;
 }
 
+// ----------------------------------------------------------------------
+// ExecSpanIterator; to eventually replace ExecBatchIterator
+
+Status ExecSpanIterator::Init(const ExecBatch& batch, ValueDescr::Shape output_shape,
```

Comment:
FTR, it's a bit confusing that the argument is called "output_shape".

Reply:
This argument (and `ValueDescr::Shape` altogether) is going to disappear soon as a result of ARROW-16757